src/Adapter/Alias/CommandHandler/ToggleAliasHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Alias\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Alias\Repository\AliasRepository;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Command\ToggleAliasCommand;
+use PrestaShop\PrestaShop\Core\Domain\Alias\CommandHandler\ToggleAliasHandlerInterfaces;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\AliasException;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\CannotToggleAliasException;
+use PrestaShopException;
+
+/**
+ * Toggles alias status
+ */
+class ToggleAliasHandler implements ToggleAliasHandlerInterfaces
+{
+    /**
+     * @var AliasRepository
+     */
+    private $aliasRepository;
+
+    /**
+     * @param AliasRepository $aliasRepository
+     */
+    public function __construct(
+        AliasRepository $aliasRepository
+    ) {
+        $this->aliasRepository = $aliasRepository;
+    }
+
+    /**
+     * @param ToggleAliasCommand $command
+     *
+     * @return void
+     *
+     * @throws AliasException
+     */
+    public function handle(ToggleAliasCommand $command): void
+    {
+        $alias = $this->aliasRepository->get($command->getAliasId());
+
+        try {
+            if (false === $alias->toggleStatus()) {
```

Comment:
instead of toggle status, use something like
$alias->status = $command->getNewStatus();
$this->aliasRepository->update($alias)

Reply:
and you won't need to wrap it in try-catch anymore as the repository will do it already (reuse AbstractObjectModelRepository)